src/TufValidatedComposerRepository.php

```diff
@@ -157,17 +157,14 @@ protected function configurePackageTransportOptions(PackageInterface $package):
         parent::configurePackageTransportOptions($package);
 
         $options = $package->getTransportOptions();
-        $config = $this->getRepoConfig();
-        // Store the information identifying this package to TUF in a format
-        // that can be safely saved to and loaded from the lock file.
-        // @see \Tuf\ComposerIntegration\Plugin::postFileDownload()
+        ['url' => $url] = $this->getRepoConfig();
+
+        // Store the information identifying this package to TUF in a form that can be stored in the lock file.
+        // @see \Tuf\ComposerIntegration\Plugin::getRepositoryFromEvent()
         $options['tuf'] = [
-            'repository' => $config['url'],
+            'repository' => $url,
             'target' => $package->getName() . '/' . $package->getVersion(),
         ];
-        if ($this->isTufEnabled($package)) {
-            $options['max_file_size'] = $this->updater->getLength($options['tuf']['target']);
-        }
```

Comment:
I don't know if this being overly paranoid but one thing we could do here is 
`$options['max_file_size'] = 0;`

with a `@see \Tuf\ComposerIntegration\TufValidatedComposerRepository::preparePackage`

Basically ensuring that if the package is TUF enabled the file size would have to set by the plugin before it could be downloaded

Reply:
I kinda like that. If it passes tests and the test project OKs it, then I'm down.

Reply:
Actually, I'm going to not do this for now, because I realized that if you set max_file_size in `configureTransportOptions()`, it's going to be stored in the lock file. I don't want a bogus file size of 0 to be stored in the lock file, in case someone installs a package _with_ TUF protection, then subsequently _disables_ TUF protection. Not sure what that would do, but since it's an unknown, it's probably better not to go there in this PR.